res/controllers/Reloop-Jockey-3-ME-scripts.js

```diff
@@ -0,0 +1,444 @@
+      newVal = curVal - interval;
+      if (newVal < min) newVal = min;
+    }
+  }
+  switch (engine.getValue("[EffectRack1_EffectUnit" + currentDeck + "_Effect1]", "num_parameters")) {
+    case 1:
+      if (EncoderKnopFX > 1) EncoderKnopFX = 0;
+      break;
+    case 2:
+      if (EncoderKnopFX > 2) EncoderKnopFX = 0;
+      break;
+  }
+  if (EncoderKnopDryWet) {
+    engine.setParameter("[EffectRack1_EffectUnit" + currentDeck + "]", "mix", newVal);
+  } else if (EncoderKnopFX) {
+    engine.setParameter("[EffectRack1_EffectUnit" + currentDeck + "_Effect1]", "parameter" + EncoderKnopFX, newVal);
+  };
+
+  // Leds
+  status -= 32;
+  if (EncoderKnopFX) {
+    Jockey3ME.effectParamLedSet(currentDeck, EncoderKnopFX, status, control);
+  } else if (EncoderKnopDryWet) {
+    Jockey3ME.effectMixLedSet(currentDeck,status,control);
+  };
+}
+
+Jockey3ME.effectParamLedSet = function (currentDeck, index, status, control) {
+  var ledValue = engine.getParameter("[EffectRack1_EffectUnit" + currentDeck + "_Effect1]", "parameter" + index);
+  ledValue = parseInt(ledValue * 127);
+  midi.sendShortMsg(status,control,ledValue);
+}
+
+Jockey3ME.effectMixLedSet = function (currentDeck, status, control) {
+  var ledValue = engine.getParameter("[EffectRack1_EffectUnit" + currentDeck + "]", "mix");
+  ledValue = parseInt(ledValue * 127);
+  midi.sendShortMsg(status,control,ledValue);
+}
+
+Jockey3ME.effectSelectLedSetNumEffect = function (currentDeck, status, control, value) {
+  // var ledValue = engine.getValue("[EffectRack1_EffectUnit" + currentDeck + "]", "num_effects");
+  var ledValue = Jockey3ME.effectsAvailable; // num_effects returns how many effects in Unit is, not what is available. Set to 5
+  if ((Jockey3ME.num_effectsValue[currentDeck - 1] + value) > ledValue) {
+    Jockey3ME.num_effectsValue[currentDeck - 1] = 1;
+  } else {
+    Jockey3ME.num_effectsValue[currentDeck - 1] += value;
+  }
+  var newLedValue = parseInt((Jockey3ME.num_effectsValue[currentDeck - 1] / ledValue) * 127);
+  midi.sendShortMsg((status-32),control,newLedValue);
+}
+
+Jockey3ME.effectSelect = function (channel, control, value, status, group) {
+  var currentDeck = parseInt(group.substring(23,24));
+  engine.setValue("[EffectRack1_EffectUnit" + currentDeck + "]", "chain_selector", (value-64));
+
+  // Set Leds
+  Jockey3ME.effectSelectTimer = engine.beginTimer(100, "Jockey3ME.effectSelectLedSet(" + status + "," + currentDeck + ")",1);
+  Jockey3ME.effectSelectLedSetNumEffect(currentDeck,status,92,(value-64));
+}
+
+Jockey3ME.effectSelectLedSet = function (status, currentDeck) {
+  status -= 32;
+  var num_parameters = engine.getValue("[EffectRack1_EffectUnit" + currentDeck + "_Effect1]", "num_parameters");
+  if (num_parameters > 3) {num_parameters = 3;};
+  if (num_parameters) {
+    for (var i = 1, j = 30; i <= num_parameters; i++) {
+      Jockey3ME.effectParamLedSet(currentDeck, i, status, j);
```

Comment:
When you change effectParamLedSet to react to changes in Mixxx with engine.connectControl, change this to use engine.trigger